tfx_addons/feature_selection/component_test.py

```diff
@@ -12,38 +12,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Tests for HelloComponent."""
-
-import json
-
-import tensorflow as tf
-from tfx.examples.custom_components.hello_world.hello_component import \
-    component
-from tfx.types import artifact, channel_utils, standard_artifacts
-
-
-class HelloComponentTest(tf.test.TestCase):
-  def setUp(self):
-    super(HelloComponentTest, self).setUp()
-    self.name = 'HelloWorld'
-
-  def testConstruct(self):
-    input_data = standard_artifacts.Examples()
-    input_data.split_names = json.dumps(artifact.DEFAULT_EXAMPLE_SPLITS)
-    output_data = standard_artifacts.Examples()
-    output_data.split_names = json.dumps(artifact.DEFAULT_EXAMPLE_SPLITS)
-    this_component = component.HelloComponent(
-        input_data=channel_utils.as_channel([input_data]),
-        output_data=channel_utils.as_channel([output_data]),
-        name=u'Testing123')
-    self.assertEqual(standard_artifacts.Examples.TYPE_NAME,
-                     this_component.outputs['output_data'].type_name)
-    artifact_collection = this_component.outputs['output_data'].get()
-    for artifacts in artifact_collection:
-      split_list = json.loads(artifacts.split_names)
-      self.assertEqual(artifact.DEFAULT_EXAMPLE_SPLITS.sort(),
-                       split_list.sort())
+"""Tests for tfx_addons.feature_selection.component"""
+
+import os
+import tempfile
+import urllib
+
+from tensorflow import test
+from tfx.components import CsvExampleGen
+from tfx.orchestration.experimental.interactive.interactive_context import \
+    InteractiveContext
+from tfx.types import standard_artifacts
+
+from tfx_addons.feature_selection import component
+
+# getting the dataset
+_data_root = tempfile.mkdtemp(prefix='tfx-data')
+DATA_PATH = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv'
+_data_filepath = os.path.join(_data_root, "data.csv")
+urllib.request.urlretrieve(DATA_PATH, _data_filepath)
+
+context = InteractiveContext()
```

Comment:
InteractiveContext only works in Notebooks!

Reply:
Oh okay, my bad. Would you have any resources for an alternate way? I wasn't able to find it as the examples I came across were implemented in Colab

@casassg

Reply:
Use LocalDagRunner instead. xgboost and sklearn examples have a test like that